pyxem/signals/diffraction2d.py

```diff
@@ -122,9 +122,21 @@ def apply_affine_transformation(
                 convert_affine_to_transform, shape=shape, inplace=False
             )
 
+        if not keep_dtype:
+            out_dtype = float
+        else:
+            out_dtype = self.data.dtype
+
+        if "output_shape" in kwargs:
+            output_shape = kwargs["output_shape"]
+        else:
+            output_shape= shape
```

Comment:
This should work (https://docs.python.org/3/library/stdtypes.html#dict.get)

```suggestion
        output_shape = kwargs.get("output_shape", shape)
```

Reply:
Huh, I didn't realize that you could do that.  Sounds like I need to go back and look though some old code now.